### PR TITLE
fix: CJS companion namespace access for resource types at all depths

### PR DESCRIPTION
**Pull request by jar-stripe**
### Why?

CJS consumers can't access nested type namespaces like `Stripe.Price.Recurring`, `Stripe.Billing.Alert.Status`, or `Stripe.Terminal.Configuration.Tipping`. The CJS declaration file uses flat type aliases inside `declare namespace StripeConstructor`, which drops companion namespaces that ESM carries automatically via `export { X }`.

Additionally, nested resource types (under product namespaces like Billing, Terminal, Apps, etc.) were broken in **both** ESM and CJS because companion namespace blocks were rendered nested under the product namespace instead of at module scope where they merge with the interface.

After this change, all type paths that work in ESM also work in CJS.

### What?

- Resource files: companion namespaces now render at module scope (`export namespace Alert { ... }`) instead of nested under the product namespace (`export namespace Billing { export namespace Alert { ... } }`)
- CJS declaration file: companion namespace blocks now emitted for resource sub-types (e.g. `Price.Recurring`, `Alert.Status`) at all depths
- CJS declaration file: params companion blocks now emitted at full depth (previously only 1 level)
- Internal references in resource files updated (e.g. `Billing.Alert.Status` → `Alert.Status`)

### See Also

- #2683
## Changelog

- Fixes TypeScript type access for nested namespaces in CJS mode (e.g. `Stripe.Price.Recurring`, `Stripe.Subscription.BillingMode`)
- Fixes TypeScript type access for nested resource types e.g. `Stripe.Billing.Alert.Status`, `Stripe.Terminal.Configuration.Tipping` in both ESM and CJS modes.

### CODE REVIEW

File: testProjects/types-cjs-node16/typescriptTest.ts [hunk 1]
```diff
@@ -39,6 +39,23 @@ async (): Promise<void> => {
 let opts: Stripe.RequestOptions;
 let apiList: Stripe.ApiList<Stripe.Customer>;
 
+// Resource sub-types (companion namespace access)
```
Comment (xavdid) [inline testProjects/types-cjs-node16/typescriptTest.ts:42]: can you add these tests to `types-cjs` as well? We should probably generate / static file copy them, but for now we just duplicate

File: src/resources/Checkout/Sessions.ts [hunk 2]
```diff
@@ -976,2475 +976,2551 @@ export interface Session {
   /**
    * Wallet-specific configuration for this Checkout Session.
    */
-  wallet_options: Checkout.Session.WalletOptions | null;
+  wallet_options: Session.WalletOptions | null;
 }
-export namespace Checkout {
```
Comment (xavdid) [inline src/resources/Checkout/Sessions.ts:981]: are these not a breaking change? or they are, but they never worked, so we're fine with it?

**Review by xavdid — APPROVED** [review]
Looks good! had a comment about more test coverage, but looks good otherwise

File: testProjects/types-cjs-node16/typescriptTest.ts (hunk 1 above)
Comment (jar-stripe) [reply testProjects/types-cjs-node16/typescriptTest.ts:42]: Ah I thought I did; it looks like it got added to types-cjs-node16.  Let me add them

File: src/resources/Checkout/Sessions.ts (hunk 2 above)
Comment (jar-stripe) [reply src/resources/Checkout/Sessions.ts:981]: My understanding is this never worked through the public API.  See 
image
captured on node v22.2.2.  This change should fix the Alert.Status red squiggle.

File: testProjects/types-cjs-node16/typescriptTest.ts (hunk 1 above)
Comment (jar-stripe) [reply testProjects/types-cjs-node16/typescriptTest.ts:42]: Fixed in [492d2dd](https://github.com/stripe/stripe-node/pull/2765/commits/492d2dd6a75603479e029ea02946e6d87195c91c)
